packages/ui/src/utils/field-validations.ts

```diff
@@ -0,0 +1,45 @@
+import i18next from 'i18next';
+
+import { ErrorType } from '@/components/ErrorMessage';
+
+const usernameRegx = /^[A-Z_a-z-][\w-]*$/;
+const emailRegEx = /^\S+@\S+\.\S+$/;
+
+export const usernameValidation = (username: string): ErrorType | undefined => {
+  if (!username) {
+    return { code: 'required', data: { field: i18next.t('input.username') } };
+  }
+
+  if (/\d/.test(username.slice(0, 1))) {
```

Comment:
better `/^\d/.test(username)`

(`^` asserts position at start of a line)

Reply:
should be fine with a single digit?

Reply:
need `^`
<img width="165" alt="image" src="https://user-images.githubusercontent.com/10806653/165020934-4695776e-9011-4a7d-bc37-065e7383470e.png">

Reply:
updated